src/luxonis_ml/data/augmentations.py

```diff
@@ -696,3 +697,72 @@ def get_params_dependent_on_targets(self, params: Dict[str, Any]) -> Dict[str, A
         target_params = super().get_params_dependent_on_targets(params)
         target_params["indices"] = list(range(self.n_tiles))
         return target_params
+
+
+@AUGMENTATIONS.register_module()
+class MixUp(A.BatchBasedTransform):
+    def __init__(
+        self,
+        alpha: float = 0.5,
+        always_apply: bool = False,
+        p: float = 0.5,
+    ):
+        """MixUp augmentation that merges two images and their annotations into one.
+        If images are not of same size then second one is first resized to match the
+        first one.
+
+        Args:
+            alpha (float, optional): Alpha value used for blending images. Defaults to 0.5.
+            always_apply (bool, optional): Defaults to False.
+            p (float, optional): Probability of applying the transform. Defaults to 1.0.
+        """
+        super().__init__(always_apply=always_apply, p=p)
+
+        self.n_tiles = 2
+        self.alpha = alpha
+        self.out_batch_size = 1
+
+    def get_transform_init_args_names(self) -> Tuple[str, ...]:
+        return ("alpha", "out_batch_size")
+
+    @property
+    def targets_as_params(self):
+        return ["image_batch"]
+
+    def apply_to_image_batch(self, image_batch, image_shapes, **params):
+        image1 = image_batch[0]
+        # resize second image to size of the first one
+        image2 = cv2.resize(image_batch[1], (image_shapes[0][1], image_shapes[0][0]))
+        img_out = cv2.addWeighted(image1, self.alpha, image2, 1 - self.alpha, 0.0)
+        return [img_out]
+
+    def apply_to_mask_batch(self, mask_batch, image_shapes, **params):
+        mask1 = mask_batch[0]
+        mask2 = cv2.resize(
```

Comment:
It might be interesting to also allow the option to paste onto one another instead of resizing to not affect aspect ratios. But I think it's fine without it for now.

Reply:
I thought about this as well. In that case we would also need to define how to align it (e.g. align top left corner, align centers of both images, etc.) - probably additional argument. But also in most dataset I think images are of same size (not in COCO though). This could be certanly added if it turns out that there is a use case for it.